modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/ObjectEntry1toMObjectRelatedModelsProviderImpl.java

```diff
@@ -68,14 +76,67 @@ public void deleteRelatedModel(
 				deletionType,
 				ObjectRelationshipConstants.DELETION_TYPE_CASCADE)) {
 
+			int maxpoolsize = GetterUtil.getInteger(
+				SystemProperties.get("system.executor.service.maxpoolsize"), 2);
+
+			ExecutorService executorService = new ThreadPoolExecutor(
+				maxpoolsize, maxpoolsize);
+
+			List<Future<?>> futures = new ArrayList<>();
+
+			String name = PrincipalThreadLocal.getName();
+			PermissionChecker permissionChecker =
+				PermissionThreadLocal.getPermissionChecker();
+
 			for (ObjectEntry objectEntry : relatedModels) {
-				_objectEntryService.deleteObjectEntry(
-					objectEntry.getObjectEntryId());
+				futures.add(
+					executorService.submit(
+						() -> {
+							try {
+								PermissionThreadLocal.setPermissionChecker(
+									permissionChecker);
+								PrincipalThreadLocal.setName(name);
+
+								_objectEntryService.deleteObjectEntry(
```

Comment:
@pedro-oliveira446 Why _objectEntryService is used here? it will do the permission checking, this could be a reason of bad performance as permission checking is involved.

I can see ObjectEntry1toMObjectRelatedModelsProviderImpl is invoked from ObjectEntryLocalServiceImpl which means it should already pass the permission checking part and reach to local service.